### PR TITLE
DRP-4203: Create Cloud Run auth python package

### PR DESCRIPTION
**Pull request by vricciardulli**
https://dealroom.atlassian.net/browse/DRP-4203

## For later PR(s)

* finish section "Best practices" in the `README.md`

* add versioning and releases

## Required checklist

Please, select all that apply. For those that do not apply, please explain why in section "Exceptions"

- [x] My code follows the [Python styleguide](https://dealroom.atlassian.net/wiki/spaces/DATA/pages/484147227/Python+Styleguide)
- [x] My code is formatted with [black](https://black.readthedocs.io/en/stable/)
- [x] I have followed the [GitHub contributing guidelines](https://dealroom.atlassian.net/wiki/spaces/DATA/pages/610828294/Contributing+guidelines+on+GitHub). In particular:
  - [x] The PR branch is in sync with the base branch
  - [x] Branch and commit names were generated using [Jira-PR-Meta-Bookmarklet](https://dealroom.atlassian.net/wiki/spaces/DATA/pages/610828294/Contributing+guidelines+on+GitHub#Jira-PR-Meta-Bookmarklet)
- [x] The local environment used to develop this code is based on `poetry.lock` or `requirements-dev.txt`
- [x] I have included the changes required by no more than 1 JIRA task
- [x] I have tested my code locally
- [x] I have checked SonarCloud and made corresponding changes were possible
- [x] I have checked that other automatic checks are OK
- [x] I have reviewed my own PR

## Optional checklist

Please, select all those that apply

- [ ] I have used new packages and I have updated `poetry.lock`, `requirements*.txt` and `pyproject.toml` accordingly
- [ ] I have made corresponding changes to the documentation 
- [ ] I have added tests and my tests are integrated in the automatic checks of this repo

## Useful links

* [Data Requests](https://dealroom.atlassian.net/jira/software/projects/DR/boards/36)
* [Data Squad 1](https://dealroom.atlassian.net/jira/software/projects/DS1/boards/37)
* [Data Squad 2](https://dealroom.atlassian.net/jira/software/projects/DS2/boards/38)

This PR was made with 💚

### CODE REVIEW

File: dealroom_cloud_run_auth/__main__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+)
+
+
+def get_default_id_token(request: google.auth.transport.Request) -> str | None:
+    creds = get_default_id_token_credentials(target_audience=None)
+    creds.refresh(request=request)
+    return creds.token
+
+
+print(get_default_id_token(requests.Request()))
```
Comment (GuillermoDR) [inline dealroom_cloud_run_auth/__main__.py:15]: is this print necessary?

File: dealroom_cloud_run_auth/session.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+def default_url_joiner(base_url: str, path: str) -> str:
+    """Use `urllib.parse.urljoin` to join the base URL and the path."""
+    return urljoin(base_url + "/", path)
+
+
+def default_user_agent() -> str | None:
+    k_service = os.getenv("K_SERVICE")
+    k_revision = os.getenv("K_REVISION")
+    if k_service and k_revision:
+        return f"{k_service}/{k_revision}"
```
Comment (GuillermoDR) [inline dealroom_cloud_run_auth/session.py:27]: wouldn't "/" be a problematic char for user-agent naming?

**Review by GuillermoDR — COMMENTED** [review]
❤️  love it

File: dealroom_cloud_run_auth/session.py (hunk 2 above)
Comment (vricciardulli) [reply dealroom_cloud_run_auth/session.py:27]: No :) just discovered that's the usual syntax for User-Agents https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent#syntax

File: dealroom_cloud_run_auth/__main__.py (hunk 1 above)
Comment (vricciardulli) [reply dealroom_cloud_run_auth/__main__.py:15]: File `__main__.py` is only executed if you do:

```shell
python -m dealroom_cloud_run_auth
```

which means that you use the module like a CLI tool on the command line to generate and **_print_** the token to stdout.

Put another way: contents of `__main__.py` are equivalent to the block of code that goes after:
```python
if __name__ == "__main__":
    # Some code
    # and more code
    # ...
```

Comment (GuillermoDR) [reply dealroom_cloud_run_auth/__main__.py:15]: clear, thanks!

**Review by GuillermoDR — APPROVED** [review]

**Pull request merged**